File: design_handoff_nogal/backend/NogalApi/Models/Product.cs
namespace NogalApi.Models;

public class Product
{
    public int Id { get; set; }
    public string Slug { get; set; } = string.Empty;
    public string Nombre { get; set; } = string.Empty;
    public string Categoria { get; set; } = string.Empty;
    public string Material { get; set; } = string.Empty;
    public decimal PrecioCOP { get; set; }
    public string? Medidas { get; set; }
    public string? Peso { get; set; }
    public string? Armado { get; set; }
    public string? Descripcion { get; set; }
    public string Estado { get; set; } = "Disponible";
    public string? ImagenUrl { get; set; }
    public bool Activo { get; set; } = true;
    public DateTime FechaCreacion { get; set; } = DateTime.UtcNow;
    public DateTime FechaActualizacion { get; set; } = DateTime.UtcNow;

    public List<ProductImage> Imagenes { get; set; } = new();
}

File: design_handoff_nogal/frontend/nogal-web/src/types/product.ts
export interface ProductImage {
  id: number
  url: string
  orden: number
}

export interface Product {
  id: number
  slug: string
  nombre: string
  categoria: string
  material: string
  precioCOP: number
  medidas?: string | null
  peso?: string | null
  armado?: string | null
  descripcion?: string | null
  estado: string
  imagenUrl?: string | null
  activo: boolean
  fechaCreacion: string
  fechaActualizacion: string
  imagenes: ProductImage[]
}

export interface PublicProduct {
  id: number
  slug: string
  nombre: string
  categoria: string
  material: string
  precioCOP: number
  medidas?: string | null
  peso?: string | null
  armado?: string | null
  descripcion?: string | null
  estado: string
  imagenUrl?: string | null
  imagenes: ProductImage[]
}

export interface PagedResult<T> {
  items: T[]
  total: number
  pagina: number
  tamano: number
}

export interface CreateProductPayload {
  nombre: string
  categoria: string
  material: string
  precioCOP: number
  medidas?: string
  peso?: string
  armado?: string
  descripcion?: string
  estado?: string
  imagenUrl?: string
}

export type UpdateProductPayload = CreateProductPayload

export interface CatalogOptions {
  categorias: string[]
  materiales: string[]
  estados: string[]
}
